### PR TITLE
Issue 50 feat/UI create modal component

### PR DESCRIPTION
**Pull request by KonTra7cf**

### CODE REVIEW

File: src/components/molecules/Modal/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    });
+
+    return (
+        <>
+            <Icon
+                iconName='comment'
+                onClick={toggleModal}
+                className='comment'
+            />
+            {showModal ? (
```
Comment (Veryyapeee) [inline src/components/molecules/Modal/Modal.jsx:34]: 1. You can go with `showModal && ...`
2. I don't think that conditional rendering is good in here, I mean, we can't animate that in this case if we want to

File: src/components/molecules/Modal/Modal.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import './Modal.scss';
+import { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { Icon } from '../../atoms/IconComponent/Icon';
+
+export const Modal = ({ children }) => {
```
Comment (Veryyapeee) [inline src/components/molecules/Modal/Modal.jsx:6]: This component should be reusable, so, you are keeping the modal state in component itself, what if for example I would like to put a modal on feedScreen, and open it on onClick of for example "Edit profile" button, I can't do that with current implementation

File: src/components/molecules/Modal/Modal.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    };
+
+    useEffect(() => {
+        document.addEventListener('keydown', keyPress);
+        return () => document.removeEventListener('keydown', keyPress);
+    });
+
+    return (
+        <>
+            <Icon
```
Comment (Veryyapeee) [inline src/components/molecules/Modal/Modal.jsx:29]: This icon shouldn't be here, we want to reuse the modal with any button, not the one here

File: src/components/molecules/Modal/Modal.jsx (hunk 2 above)
Comment (Veryyapeee) [reply src/components/molecules/Modal/Modal.jsx:6]: To do that we can just pass the state and setState as a props, and use them from props, in that way we would have state in parent component and we will be able to do anything we want with modal visibility

File: src/components/molecules/Modal/Modal.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    );
+
+    const toggleModal = () => {
+        setShowModal((prev) => !prev);
+    };
+
+    useEffect(() => {
+        document.addEventListener('keydown', keyPress);
+        return () => document.removeEventListener('keydown', keyPress);
+    });
```
Comment (Veryyapeee) [inline src/components/molecules/Modal/Modal.jsx:25]: We should pass an empty dependency array in here, to make sure that this useEffect will run only twice, mount and unmount the component

File: src/components/atoms/IconComponent/Icon.scss [hunk 5]
```diff
@@ -12,5 +12,17 @@
     &__large {
```
Comment (Veryyapeee) [inline src/components/atoms/IconComponent/Icon.scss:12]: After adjustments, can you add a test modal on feed screen, just for testing purposes?

**Review by Veryyapeee — CHANGES_REQUESTED** [review]

File: src/components/molecules/Modal/Modal.jsx (hunk 4 above)
Comment (KonTra7cf) [reply src/components/molecules/Modal/Modal.jsx]: I had a warning : "React Hook useEffect has a missing dependency: 'keyPress'. Either include it or remove the dependency array. " so I put a dependency 'keyPress', but if that's a problem, I can always remove this additional 'keyPress' feature from the component.

File: src/components/atoms/IconComponent/Icon.scss (hunk 5 above)
Comment (KonTra7cf) [reply src/components/atoms/IconComponent/Icon.scss:12]: Sure - I hope I did it the right way

**Comment by Veryyapeee** [comment]
looks good to me, 1 issue, the modal kinda sticks to the bottom of the site,

**Comment by KonTra7cf** [comment]
@Veryyapeee 
I changed a positioning a bit so modal doesn't stick to the bottom anymore and I think it looks ok now

**Comment by Veryyapeee** [comment]
> @Veryyapeee I changed a positioning a bit so modal doesn't stick to the bottom anymore and I think it looks ok now

Yeah, fine for me, I think that in future we will need to adjust some designs, but for now it's hard to tell when we can use the modal only in like "dummy" way :D

**Review by T0ffik — APPROVED** [review]

**Pull request merged**